docs/UpgradeGuide2.md

```diff
@@ -0,0 +1,529 @@
+  | `profiling.advanced.max_events`                         | Removed                             | (none)                               |
+  | `profiling.advanced.legacy_transport_enabled`           | Removed                             | (none)                               |
+  | `profiling.advanced.force_enable_new_profiler`          | Removed                             | (none)                               |
+  | `profiling.advanced.force_enable_legacy_profiler`       | Removed                             | (none)                               |
+
+<h2 id="2.0-upgrading-advanced-usage">Upgrading Advanced Usage</h2>
+
+<h3 id="2.0-frozen-string">Frozen String</h3>
+
+All strings are frozen by default. Make sure your code does not mutate them.
+
+<h3 id="2.0-configuration-api">Configuration API</h3>
+
+If you are [writing your own instrumentation](DevelopmentGuide.md#writing-new-integrations),
+configuration options are now lazily evaluated by default, hence remove `.lazy` from option definition.
+
+```ruby
+class MySettings < Datadog::Tracing::Contrib::Configuration::Settings
+  option :boom do |o|
+    o.default do
+      true
+    end
+    o.lazy # === Remove this with 2.0 ===
+  end
+end
+```
+
+<h3 id="2.0-tracing-api">Tracing API</h3>
+
+Remove option `span_type` from `Datadog::Tracing.trace` method and the following alias methods
+
+| 1.x                                            | Replacement in 2.0                        |
+| ---------------------------------------------- | ----------------------------------------- |
+| `Datadog::Tracing.trace(name, span_type: ...)` | `Datadog::Tracing.trace(name, type: ...)` |
+| `Datadog::Tracing::SpanOperation#span_id`      | `Datadog::Tracing::SpanOperation#id`      |
+| `Datadog::Tracing::SpanOperation#span_type`    | `Datadog::Tracing::SpanOperation#type`    |
+| `Datadog::Tracing::SpanOperation#span_type=`   | `Datadog::Tracing::SpanOperation#type=`   |
+| `Datadog::Tracing::Span#span_id`               | `Datadog::Tracing::Span#id`               |
+| `Datadog::Tracing::Span#span_type`             | `Datadog::Tracing::Span#type`             |
+| `Datadog::Tracing::Span#span_type=`            | `Datadog::Tracing::Span#type=`            |
+
+If you are using [manual instrumentation](GettingStarted.md#manual-instrumentation) or [processing pipeline](GettingStarted.md#processing-pipeline)
+
+```ruby
+# === with 1.x ===
+Datadog::Tracing.trace('my_span', span_type: 'custom') do |span|
+  puts span.span_id
+
+  span.span_type = "...."
+end
+
+# === with 2.0 ===
+Datadog::Tracing.trace('my_span', type: 'custom') do |span|
+  puts span.id
+
+  span.type = "...."
+end
+```
+
+<h3 id="2.0-log-correlation">Log Correlation</h3>
+
+Remove obsolete fields from `Datadog::Tracing::Correlation::Identifier`, and responds to `#env`, `#service`, `#version`, `#trace_id`, `#span_id` and `#to_log_format`. The values returned from `#trace_id` and `#span_id` change from `Integer` to `String`.
```

Comment:
This paragraph could probably be made clearer. Maybe something more like "The following fields have been removed from `Datadog::Tracing::Correlation::Identifier and it no longer responds to...

I would also maybe separate `The values returned from....` into a separate paragraph.